### PR TITLE
contrib: Catch signals to delete tmp folder in local-analyze-images

### PR DESCRIPTION
**Pull request by Quentin-M**
- Create the temporary folder earlier in the app (main instead of save)
- Catch SIGINT/SIGKILL signals to delete the temporary folder and exit

### CODE REVIEW

File: contrib/analyze-local-images/main.go [hunk 1]
```diff
@@ -110,24 +111,40 @@ func main() {
...
+	go func() {
+		<-interrupt
+		os.RemoveAll(tmpPath)
+		os.Exit(130)
+	}()
+
+	// Analyze the image.
+	err = AnalyzeLocalImage(imageName, minSeverity, *flagEndpoint, *flagMyAddress, tmpPath)
+	if err != nil {
+		log.Print(err)
```
Comment (jzelinskie) [inline contrib/analyze-local-images/main.go]: this case is going to error, but exit 0

**Comment by jzelinskie** [comment]
Let's wrap main so that we can return custom errors and get custom exit codes.

``` go
package main

import "fmt"
import "os"

func doTheStuff() int {
    defer fmt.Println("!")

    return 3
}

func main() {
    os.Exit(doTheStuff())
}
```

Let's also block on a select at the bottom of the function waiting for analyzelocalimage or the interrupt

**Comment by Quentin-M** [comment]
Meh ^

**Comment by jzelinskie** [comment]
LGTM

**Pull request merged**